src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/commands/fork/index.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ESQLCommand } from '@kbn/esql-ast';
+import { i18n } from '@kbn/i18n';
+import {
+  getCommandDefinition,
+  getCommandsByName,
+  pipePrecedesCurrentWord,
+} from '../../../shared/helpers';
+import { CommandSuggestParams } from '../../../definitions/types';
+import type { SuggestionRawDefinition } from '../../types';
+import { TRIGGER_SUGGESTION_COMMAND } from '../../factories';
+import { suggest as suggestForWhere } from '../where';
+import { suggest as suggestForSort } from '../sort';
+import { suggest as suggestForLimit } from '../limit';
+import { getCommandAutocompleteDefinitions, pipeCompleteItem } from '../../complete_items';
+
+export async function suggest(
+  params: CommandSuggestParams<'fork'>
+): Promise<SuggestionRawDefinition[]> {
+  if (/FORK\s+$/i.test(params.innerText)) {
+    return [newBranchSuggestion];
+  }
+
+  const activeBranch = getActiveBranch(params.command);
+  const withinActiveBranch =
+    activeBranch &&
+    activeBranch.location.min <= params.innerText.length &&
+    activeBranch.location.max >= params.innerText.length;
+  if (!withinActiveBranch) {
+    // not within a branch
+    if (/\)\s+$/i.test(params.innerText)) {
+      return [newBranchSuggestion, pipeCompleteItem];
+    }
+  }
+
+  // within a branch
+  if (activeBranch?.commands.length === 0 || pipePrecedesCurrentWord(params.innerText)) {
+    return getCommandAutocompleteDefinitions(getCommandsByName(['limit', 'sort', 'where']));
+  }
+
+  const subCommand = activeBranch?.commands[activeBranch.commands.length - 1];
+  switch (subCommand?.name) {
+    case 'where':
+      return suggestForWhere({
+        ...params,
+        command: subCommand as ESQLCommand<'where'>,
+        definition: getCommandDefinition('where'),
+      });
+
+    case 'sort':
+      return suggestForSort({
+        ...params,
+        command: subCommand as ESQLCommand<'sort'>,
+        definition: getCommandDefinition('sort'),
+      });
+
+    case 'limit':
+      return suggestForLimit({
+        ...params,
+        command: subCommand as ESQLCommand<'limit'>,
+        definition: getCommandDefinition('limit'),
+      });
```

Comment:
nit: These all look very similar. It is now that only WHERE, SORT, and LIMIT commands are allowed; maybe in the future there will be more commands? In any case, it is the validation/parsing that can detect unexpected commands, but here&mdash;at the autocomplete stage&mdash;we can maybe just pass the autocomplete logic to the respective commands? We be cool to have (or maybe reuse, if already exists) some `superSuggest` function, which just routes to the right command:

```suggestion
      return superSuggest`({
        ...params,
        command: subCommand,
        definition: getCommandDefinition(subCommand?.name),
      });
```

Reply:
Yeah, I think this makes sense.

Reply:
More commands will be allowed so the more dynamic our code is the better

Reply:
Here they are! More commands https://github.com/elastic/elasticsearch/pull/125937

Reply:
Done in https://github.com/elastic/kibana/pull/216743/commits/712748eb627fa0cbb29f62368255d467e2db2984